Gui/CurveWidget.cpp

```diff
@@ -164,6 +164,7 @@ CurveWidget::initializeGL()
     // always running in the main thread
     assert( qApp && qApp->thread() == QThread::currentThread() );
     appPTR->initializeOpenGLFunctionsOnce();
+    makeCurrent();
```

Comment:
I don't understand why you call makeCurrent here.

From the [doc](https://doc.qt.io/qt-6/qopenglwidget.html#initializeGL): There is no need to call [makeCurrent](https://doc.qt.io/qt-6/qopenglwidget.html#makeCurrent)() because this has already been done when this function is called.

Reply:
So these calls are needed because initializeOpenGLFunctionsOnce() can end up creating a context, making it current, and then does not restore the old context. I believe we should fix that behavior instead of papering over it here and below. I started looking into that but haven't got a full solution yet.  I also am a little skeptical about the need to init all the GL functions in these various places. ISTM that this should happen very early in startup instead of here, but I'm not as familiar with this code yet.

I suspect making something like my [ScopedGLContext](https://github.com/NatronGitHub/Natron/blob/RB-2.6/Engine/OSGLContext_win.cpp#L733C8-L733C22) on Windows cross-platform would make resolving this and any other similar situation we encounter simpler.

Reply:
I agree, I thought it was a good idea to push what I found to help maintainers debug the code.